src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -159,7 +161,13 @@ public JsonClassInfo(Type type, JsonSerializerOptions options)
 
                         if (converter.ConstructorIsParameterized)
                         {
-                            converter.CreateConstructorDelegate(options);
+                            // Create and cache the constructor delegate for this type.
+                            // We depend on the first converter created for this type to perform the initialization.
```

Comment:
Is the first converter created and this method called (delegate assigned) before the converter instance is added to the cache on the JsonSerializerOptions instance? If not, then it is possible to return the converter with no delegate which may be used by a different thread (and likely cause other issues).

Reply:
Right now, the delegate assignment is done after the converter is added to the cache. I'll push a commit moving this assignment to the start of deserialization (one extra null check) - this should be the safest option.